src/main/java/com/palantir/gradle/versions/GradleWorkarounds.java

```diff
@@ -51,6 +58,37 @@ static boolean isConfiguring(ProjectState state) {
         }
     }
 
+    /**
+     * Work around gradle < 5.3.rc-1 not adding an AttributeFactory to {@link ProjectDependency} with configuration,
+     * and {@link ExternalDependency#copy()} not configuring an AttributeFactory and ALSO not immutably copying the
+     * {@link AttributeContainer}.
+     */
+    static <T extends ModuleDependency> T fixAttributesOfModuleDependency(
```

Comment:
the reflective parts of this method can go away once this PR goes in: https://github.com/gradle/gradle/pull/9653